clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -328,6 +343,14 @@ public synchronized Set<String> subscription() {
         return Collections.emptySet();
     }
 
+    public synchronized SubscriptionPattern subscriptionPattern() {
+        return this.subscriptionPattern;
+    }
+
+    public synchronized Pattern subscribedPattern() {
+        return this.subscribedPattern;
+    }
+
```

Comment:
Super nit: I've seen committers request that we drop unnecessary `this` qualifiers:

```suggestion
    public synchronized SubscriptionPattern subscriptionPattern() {
        return subscriptionPattern;
    }

    public synchronized Pattern subscribedPattern() {
        return subscribedPattern;
    }

```